### ISSUE TITLE
Fix tools writing to Galaxy's files directory

### ISSUE THREAD

**Issue by bernt-matthias** [galaxyproject/tools-iuc#3991]
With https://github.com/galaxyproject/planemo/pull/1190 its possible to detect tools that write to Galaxy's files directory. I'm running this on IUC here: https://github.com/bernt-matthias/tools-iuc/runs/3721212988?check_suite_focus=true

See also discussion in https://github.com/galaxyproject/tools-iuc/pull/3986#issuecomment-926729276

Here is a list of tools (from an earlier run of the CI workflow) that create at least one extra file: 

- [ ] emboss	.dat.1.png
- [ ] bacteria_tradis	.dat.bwt ...
- [ ] barrnap	.dat.fai https://github.com/galaxyproject/tools-iuc/pull/4104
- [ ] commet	.dat_multi_k31_s4.h5
- [ ] crossmap_*	.dat.fai	.unmap https://github.com/galaxyproject/tools-iuc/pull/4126
- [ ] delly_*	.dat.fai
- [ ] discosnp_pp	.dat_multi_k31_s4.h5
- [ ] fermikit https://github.com/galaxyproject/tools-iuc/pull/4105
  - fermi2	.dat.fai
  - fermikit_variants	.dat.fai
- [ ] hyphy_*	.dat.fit.bf
- [ ] irissv	.dat.bai https://github.com/galaxyproject/tools-iuc/pull/4642 `1.0.4+galaxy2`
- [ ] kissplice	.dat_multi_k31_s4.h5
- [ ] lordec	.dat_multi_k31_s4.h5
- [ ] mapsembler2	.dat_multi_k31_s4.h5
- [ ] medaka_*	.dat.fai ...
- [ ] nanocompore_*	.dat.fai
- [ ] picard_*	.dat.bai
- [ ] progressivemauve	.dat.backbone
- [ ] pyega3	.dat.md5
- [ ] stringtie*	.dat.bak
- [ ] takeabreak	.dat_multi_k31_s4.h5
- [ ] tetyper	.dat.nto ... https://github.com/galaxyproject/tools-iuc/pull/4107
- [ ] tradis_*	.dat.bwt .. https://github.com/galaxyproject/tools-iuc/pull/4103
- [ ] xmfa2gff3	.dat.backbone

I hope that the PR is accepted soon. Then it will also run for PRs.

Put your name / PR number next to the entries if you are working on one of the tools above.

**Comment by natefoo** [comment]
It'd be great if we could also record version numbers where this was fixed, which will make admins' lives much easier.

**Comment by bernt-matthias** [comment]
> It'd be great if we could also record version numbers where this was fixed, which will make admins' lives much easier.

Any suggestion how to achieve this? Just a txt file, maybe in a gist?

**Comment by natefoo** [comment]
Yeah, I don't have a strong preference - gist, or just edit your comment in this issue to record it next to the tool ID as it is checked off?

**Comment by bernt-matthias** [comment]
This should now be covered by the containerized tool tests.
